vscripts/confogl.nut

```diff
@@ -2,69 +2,244 @@
+  Msg(format("CanPickupObject(%s)\n", object.tostring()));
+  // Default to disallow the pickup.
+  local retVal = false;
+  runPluginHookCallbacks("CanPickupObject", function (hookFn) {
+    local res = hookFn(object);
+    // If any plugin returns true, we return true.
+    if (res == true) {
+      return true;
+    }
+  });
+  return retVal;
+}
+// void InterceptChat(string message, CTerrorPlayer speaker): If you put a function with this name in your script, C++ will call it on all chat messages. Passing in the (annotated) chat string and the handle of the speaker.
+// If the function returns false, it will prevent other players from seeing the chat message except for the player who entered it. 
+function InterceptChat(message, speaker) {
+  Msg(format("InterceptChat(%s, %s)\n", message, speaker.tostring()));
+  // Default to allow the message to be seen by other players.
+  local retVal = true;
+  runPluginHookCallbacks("InterceptChat", function (hookFn) {
+    local res = hookFn(message, speaker);
+    // If any plugin returns false, we return false.
+    if (res == false) {
+      return false;
+    }
+  });
+  return retVal;
+}
+// void UserConsoleCommand(handle playerScript, arg): when a user does a <scripted_user_func argument> at console (or bound to a key) this function is called (if it exists).
+// The playerscript is which players console it came from. You can pass strings or whatever, of course. So could do a switch statement off <arg> to give players special controls, etc. 
+function UserConsoleCommand(playerScript, arg) {
+  Msg(format("UserConsoleCommand(%s)\n", playerScript.tostring()));
+  // Default to allow the command(? Does returning false disallow it?).
+  local retVal = true;
+  runPluginHookCallbacks("UserConsoleCommand", function (hookFn) {
+    local res = hookFn(playerScript, arg);
+    // If any plugin returns false, we return false.
+    if (res == false) {
+      return false;
+    }
+  });
+  return retVal;
+}
+
+function OnGameplayStart() {
+  // A callback that is called when all players have spawned and players start to play.
+  unPluginHookCallbacks("OnGameplayStart", function (hookFn) {
+    hookFn();
+  });
+}
+
+// Update gets called much like a Think() function.
+function Update() {
+  runPluginHookCallbacks("Update", function (hookFn) {
+    hookFn();
+  });
+}
+
+DirectorOptions <- {
   // bool AllowFallenSurvivorItem(string classname)
   // Returns true or false if Fallen Survivors are allowed to carry the given classname.
   function AllowFallenSurvivorItem(classname) {
     Msg(format("DirectorOptions.AllowFallenSurvivorItem(%s)\n", classname));
-    return true;
+    // Default to disallow carrying the classname.
+    local retVal = false;
+    g_ModeScript.runPluginHookCallbacks("DirectorOptions.AllowFallenSurvivorItem", function (hookFn) {
+      local res = hookFn(classname);
+      // If any plugin returns true, we return true.
+      if (res == true) {
+        return true;
```

Comment:
These `return false`s aren't going to work anymore since we're inside a callback function here. We need to modify the retVal instead, or add some short circuiting logic to `runPluginHookCallbacks`

Reply:
Done: https://github.com/Icy-Inferno/confogl-mutation/commit/b1bfeaaa111ea3ac3e72e79f928697e6fad9f0c1